grpc/estimator_service.cc

```diff
@@ -339,26 +358,110 @@ grpc::Status EstimatorService::Weights(grpc::ServerContext* context,
   output->set_prior(estimator_.scale_prior_);
 
   // sensor
-  int ns = estimator_.dim_sensor_;
-  if (input.sensor_size() == ns) {
-    mju_copy(estimator_.scale_sensor_.data(), input.sensor().data(), ns);
+  int num_sensor = estimator_.num_sensor_;
+  if (input.sensor_size() == num_sensor) {
+    mju_copy(estimator_.scale_sensor_.data(), input.sensor().data(),
+             num_sensor);
   }
-  for (int i = 0; i < ns; i++) {
+  for (int i = 0; i < num_sensor; i++) {
     output->add_sensor(estimator_.scale_sensor_[i]);
   }
 
   // force
-  int nv = estimator_.model_->nv;
-  if (input.force_size() == nv) {
-    mju_copy(estimator_.scale_force_.data(), input.force().data(), nv);
+  int num_jnt = 4;
+  if (input.force_size() == num_jnt) {
+    mju_copy(estimator_.scale_force_.data(), input.force().data(), num_jnt);
   }
-  for (int i = 0; i < nv; i++) {
+  for (int i = 0; i < num_jnt; i++) {
     output->add_force(estimator_.scale_force_[i]);
   }
 
   return grpc::Status::OK;
 }
 
+grpc::Status EstimatorService::Norms(grpc::ServerContext* context,
+                                     const estimator::NormRequest* request,
+                                     estimator::NormResponse* response) {
+  if (!Initialized()) {
+    return {grpc::StatusCode::FAILED_PRECONDITION, "Init not called."};
+  }
+
+  // norm
+  estimator::Norm input = request->norm();
+  estimator::Norm* output = response->mutable_norm();
+
+  // set sensor type
+  int num_sensor = estimator_.num_sensor_;
+  if (input.sensor_type_size() == num_sensor) {
```

Comment:
if the python user made a mistake and got the number of sensors wrong, they may not understand why their call, intended as "set", isn't having the right effect.  Maybe we should bite the bullet and stick to the standard of having two proto calls, one for set and one for get?